### PR TITLE
Update to version 2.0: First Task - Remove ExtendedQueryRoot

### PR DESCRIPTION
**Pull request by CCheSumo**
- Removed ExtendedQueryRoot
- Unified sortOpt in QueryRoot
- Some minor code clean up

This is a breaking change. Needs version 2.0

### CODE REVIEW

**Comment by rcoh** [comment]
Can you fork off a 1.x branch if you haven't already? We need a place to make bug fixes against the 1.0 client until 2.0 is ready.

**Comment by CCheSumo** [comment]
Ah, sure. The fork branch is release-1.0.x-bug-fixes. We can cherry-pick to this branch all the fixes needed.

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala [hunk 1]
```diff
@@ -63,34 +64,12 @@ trait QueryDsl extends DslCommons with SortDsl {
     def apply(query: Query,
```
Comment (rcoh) [inline elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala]: do we still need apply even when we don't need backwards compat?

**Review by rcoh — APPROVED** [review]

File: elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala (hunk 1 above)
Comment (CCheSumo) [reply elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala]: This is to supply default of None to optional args. I guess we can do that in the class interface without apply. Updating

**Comment by rcoh** [comment]
@CCheSumo seems like you merged this into master? I thought we had discussed merging into a 2.0 branch..?

**Comment by CCheSumo** [comment]
@rcoh I made a branch for 1.0.x branch and was cherry picking to it. But yeah, making a 2.0 branch and keep main branch at 1.0 makes sense. Reverting this and abandoning the 1.0.x branch.

**Pull request merged**